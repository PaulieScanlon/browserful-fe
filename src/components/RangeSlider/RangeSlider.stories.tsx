import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import { RangeSlider } from './RangeSlider';
import { colours } from '../../theme';

const stories = storiesOf('RangeSlider', module);

const onChange = value => {
  action('onChange')('min: ', value[0], 'max: ', value[1]);
};

const styles = {
  marginTop: '25px'
};

stories.add(
  'default usage',
  withInfo(
    'RangeSlider accepts input ranges for versions, global usage and release dates'
  )(() => (
    <div style={styles}>
      <RangeSlider min={1970} max={2018} onChange={value => onChange(value)} />
    </div>
  ))
);

stories.add(
  'sliderColour: teal',
  withInfo('RangeSlider can be themed by using the sliderColour prop')(() => (
    <div style={styles}>
      <RangeSlider
        min={1970}
        max={2018}
        sliderColour={colours.teal}
        onChange={value => onChange(value)}
      />
    </div>
  ))
);

stories.add(
  'sliderColour: blue',
  withInfo('RangeSlider can be themed by using the sliderColour prop')(() => (
    <div style={styles}>
      <RangeSlider
        min={1970}
        max={2018}
        sliderColour={colours.blue}
        onChange={value => onChange(value)}
      />
    </div>
  ))
);
