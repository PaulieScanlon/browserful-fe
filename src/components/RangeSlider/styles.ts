import styled, { css } from 'react-emotion';
import { colours, common } from '../../theme';
import { spaceSm } from '../../theme/index';

const colourStyles = props =>
  css`
    .rc-slider-track {
      background-color: ${props.sliderColour};
    }

    .rc-slider-handle {
      background-color: ${props.sliderColour};
    }
  `;

const coreStyles = () =>
  css`
    .rc-slider {
      height: 24px;
      padding: 10px 0px;
      width: 95%;
      position: relative;
      margin: 0 auto;
    }

    .rc-slider-rail,
    .rc-slider-track,
    .rc-slider-step {
      height: 6px;
    }

    .rc-slider-rail {
      background-color: ${colours.greyLight};
      border-radius: 0px;
    }

    .rc-slider-handle {
      width: 22px;
      height: 22px;
      margin-top: -9px;
      margin-left: -12px;
      border-color: ${colours.offWhite};
      box-shadow: ${common.materialBoxShadow1};
    }

    .rc-slider-dot {
      display: none;
    }
    .rc-slider-mark-text {
      margin-top: ${spaceSm}px;
    }
  `;

export const RangeSliderWrapper = styled('span')`
  ${coreStyles};
  ${colourStyles};
  position: relative;
  display: block;
  margin: 10px 0px 40px 0px;
`;
