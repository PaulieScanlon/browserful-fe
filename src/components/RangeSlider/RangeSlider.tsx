import * as React from 'react';
import Tooltip from 'rc-tooltip';
import Slider from 'rc-slider';
const d3 = require('d3-scale');

import 'rc-slider/assets/index.css';
import 'rc-tooltip/assets/bootstrap.css';

const createSliderWithTooltip = Slider.createSliderWithTooltip;
const Range = createSliderWithTooltip(Slider.Range);
const Handle = Slider.Handle;

import { RangeSliderWrapper } from './styles';
import { colours } from '../../theme';

interface IProps {
  min: number;
  max: number;
  sliderColour?: string;
  onChange: (value: number[]) => void;
}

const marks = {
  1970: '1970',
  2018: 'now'
};

const createMarks = (min: number, max: number) => {
  const x = d3
    .scaleLinear()
    .domain([min, max])
    .range([min, max]);

  return x.ticks(9);
};

const RangeHandle: React.SFC<{}> = (props): any => {
  const { value, dragging, index }: any = props;

  return (
    <Tooltip visible={dragging} placement="top" key={index}>
      <Handle value={value} />
    </Tooltip>
  );
};

export const RangeSlider: React.SFC<IProps> = ({
  min,
  max,
  sliderColour,
  onChange
}: IProps) => {
  createMarks(min, max);

  return (
    <RangeSliderWrapper sliderColour={sliderColour}>
      <Range
        min={min}
        max={max}
        defaultValue={[min, max]}
        handle={RangeHandle}
        marks={marks}
        // tipFormatter={value => `${value}%`}
        tipFormatter={value => value}
        // tipProps={{
        //   visible: true
        // }}
        onChange={onChange}
      />
    </RangeSliderWrapper>
  );
};

RangeSlider.defaultProps = {
  sliderColour: colours.pink
};
