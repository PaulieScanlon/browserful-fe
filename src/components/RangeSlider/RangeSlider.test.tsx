import * as React from 'react';
import { shallow } from 'enzyme';

import { RangeSlider } from './RangeSlider';

const onChange = event => {};

const shallowDefault = props => {
  return shallow(
    <RangeSlider
      {...props}
      min={1970}
      max={2018}
      steps={8}
      onChange={event => onChange(event)}
    />
  );
};

describe('<RangeSlider>', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault({});
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
