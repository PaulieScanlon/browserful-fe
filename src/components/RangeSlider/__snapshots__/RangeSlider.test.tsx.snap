// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeSlider> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RangeSlider
    max={2018}
    min={1970}
    onChange={[Function]}
    sliderColour="#ff5ca3"
    steps={8}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ComponentWrapper
        defaultValue={
          Array [
            1970,
            2018,
          ]
        }
        handle={[Function]}
        handleStyle={
          Array [
            Object {},
          ]
        }
        marks={
          Object {
            "1970": 1970,
            "1978": 1978,
            "1986": 1986,
            "1994": 1994,
            "2002": 2002,
            "2010": 2010,
            "2018": 2018,
          }
        }
        max={2018}
        min={1970}
        onChange={[Function]}
        tipFormatter={[Function]}
        tipProps={Object {}}
      />,
      "sliderColour": "#ff5ca3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultValue": Array [
          1970,
          2018,
        ],
        "handle": [Function],
        "handleStyle": Array [
          Object {},
        ],
        "marks": Object {
          "1970": 1970,
          "1978": 1978,
          "1986": 1986,
          "1994": 1994,
          "2002": 2002,
          "2010": 2010,
          "2018": 2018,
        },
        "max": 2018,
        "min": 1970,
        "onChange": [Function],
        "tipFormatter": [Function],
        "tipProps": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ComponentWrapper
          defaultValue={
            Array [
              1970,
              2018,
            ]
          }
          handle={[Function]}
          handleStyle={
            Array [
              Object {},
            ]
          }
          marks={
            Object {
              "1970": 1970,
              "1978": 1978,
              "1986": 1986,
              "1994": 1994,
              "2002": 2002,
              "2010": 2010,
              "2018": 2018,
            }
          }
          max={2018}
          min={1970}
          onChange={[Function]}
          tipFormatter={[Function]}
          tipProps={Object {}}
        />,
        "sliderColour": "#ff5ca3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultValue": Array [
            1970,
            2018,
          ],
          "handle": [Function],
          "handleStyle": Array [
            Object {},
          ],
          "marks": Object {
            "1970": 1970,
            "1978": 1978,
            "1986": 1986,
            "1994": 1994,
            "2002": 2002,
            "2010": 2010,
            "2018": 2018,
          },
          "max": 2018,
          "min": 1970,
          "onChange": [Function],
          "tipFormatter": [Function],
          "tipProps": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
